sklearn/ensemble/_hist_gradient_boosting/predictor.py

```diff
@@ -30,21 +30,28 @@ def get_max_depth(self):
         """Return maximum depth among all leaves."""
         return int(self.nodes['depth'].max())
 
-    def predict(self, X):
+    def predict(self, X, X_binned_cat=None, orig_feature_to_binned_cat=None):
         """Predict raw values for non-binned data.
 
         Parameters
         ----------
         X : ndarray, shape (n_samples, n_features)
             The input samples.
+        X_binned_cat : ndarray, shape (n_samples, n_categorical_features), \
+            default=None
+            Binned category features.
+        orig_feature_to_binned_cat : ndarray, shape (n_features), default=None
+            Mapping from originl feature index to column corresponding to
+            ``X_binned_cat``.
 
         Returns
         -------
         y : ndarray, shape (n_samples,)
             The raw predicted values.
         """
         out = np.empty(X.shape[0], dtype=Y_DTYPE)
-        _predict_from_numeric_data(self.nodes, X, out)
+        _predict_from_numeric_data(
```

Comment:
I feel like this one shouhld be called `predict_from_non_binned_data` now, because of the potential confusion with numerical vs categorical, in contrast to numerical vs binned

Reply:
Renamed to `_predict_from_data` for the lack of a better name. In its current state, it is taking two arrays, a non-binned numerical ndarray and a binned categorical ndarray (if it exist). This goes hand in hand with "do we want to bin during predict".